OSDOCS-7308: run MicroShift on disconnected host (#70065)
openshift/openshift-docs: modules/microshift-disconnected-host-con.adoc

// Module included in the following assemblies:
//
// * microshift_networking/microshift-disconnected-network-config.adoc

:_mod-docs-content-type: CONCEPT
[id="microshift-disconnected-host-preparation_{context}"]
= Preparing networking for fully disconnected hosts

> Use the procedure that follows to start and run {microshift-short} clusters on devices running fully disconnected operating systems. A {microshift-short} host is considered fully disconnected if it has no addressable subnet.

Reviewer: ```suggestion
Use the procedure that follows to start and run {microshift-short} clusters on devices running fully disconnected operating systems. A {microshift-short} host is considered fully disconnected if it has no external network connectivity.
```

The subnet description might be a little obtuse for users.  This says (more or less) the same thing without insinuating the need to go and check for a subnet.

> Typically this means that the device does not have an attached network interface controller (NIC) to provide a subnet. These steps can also be completed on a host with a NIC that is removed after setup. You can also automate these steps on a host that does not have a NIC by using the `%post` phase of a kickstart file.

Reviewer 2: ```suggestion
Typically this means that the device does not have an attached network interface controller (NIC) to provide a subnet. These steps can also be completed on a host with a NIC that is removed after setup. You can also automate these steps on a host that does not have a NIC by using the `%post` phase of a Kickstart file.
```

[IMPORTANT]
====

> Configuring networking settings for disconnected environments is necessary because {microshift-short} requires a network device to support cluster communication. To meet this requirement, you must configure {microshift-short} networking settings to use the IP address of a system loopback device.

Reviewer: ```suggestion
Configuring networking settings for disconnected environments is necessary because {microshift-short} requires a network device to support cluster communication. To meet this requirement, you must configure {microshift-short} networking settings to use the IP address we will assign to the system loopback device.
```

The IP we're hinting at here doesn't exist until it's created during the setup (later called the "fake" IP addr).

====

[id="microshift-disconnected-host-procedure-summary{context}"]
== Procedure summary

To run {microshift-short} on a disconnected host, the following steps are required:

Prepare the host::
* Stop {microshift-short} if it is currently running and clean up changes the service has made to the network.
* Set a persistent hostname.
* Add a “fake” IP address on the loopback interface.
* Configure DNS to use the fake IP as local name server.
* Add an entry for the hostname to `/etc/hosts`.

Update the {microshift-short} configuration::

> * Define the `nodeIP` parameter as the new loopback IP address.

Reviewer: ```suggestion
* Define the `.node.nodeIP` parameter as the new loopback IP address.
```

> * Set the `hostnameOverride` parameter to the persistent hostname.

Reviewer: ```suggestion
* Set the `.node.hostnameOverride` parameter to the persistent hostname.
```
